src/tests/multihost/alltests/test_krb_access_provider.py

```diff
@@ -185,22 +180,18 @@ def test_0003_k5login_user3_with_user4(multihost):
         for command in echo_cmds:
             cmd = multihost.client[0].run_command(command, raiseonerr=False)
 
-        user = 'foo3'
-        ssh_output = client.auth_from_client(user, 'Secret123')
-
+        check_login_client(multihost, "foo3", "Secret123")
         client_hostname = multihost.client[0].sys_hostname
-        ssh = pexpect_ssh(client_hostname, 'foo4', 'Secret123', debug=False)
-        ssh.login(login_timeout=10, sync_multiplier=1, auto_prompt_reset=False)
-        ssh.command(f"ssh -o StrictHostKeyChecking=no -o PasswordAuthentication=no foo3@{client_hostname}\
-                     'id > /tmp/accessProvider_id_krb5_003.out 2>&1' ")
-        ssh.logout()
+        run_command_client(multihost, "foo4", "Secret123",
+                           f"ssh -o StrictHostKeyChecking=no -o "
+                           f"PasswordAuthentication=no foo3@{client_hostname} "
+                           f"id > /tmp/accessProvider_id_krb5_003.out 2>&1")
 
         logfile = multihost.client[0].get_file_contents("/tmp/accessProvider_id_krb5_003.out").decode('utf-8')
 
         multihost.client[0].run_command("truncate -s 0 /home/foo3/.k5login", raiseonerr=False)
```

Comment:
This (teardown) will not be executed when the check_login_client fails on exception.
The whole point in putting asserts at the end of the test is to  do the cleanup before evaluation/failure